compiler/rustc_codegen_ssa/src/back/link.rs

```diff
@@ -178,6 +195,8 @@ pub fn link_binary(
                 tempfiles_for_stdout_output.push(out_filename);
             }
         }
+
+        check_link_info_print_request(sess, &codegen_results.crate_info.crate_types);
     }
```

Comment:
```suggestion
    }

    check_link_info_print_request(sess, &codegen_results.crate_info.crate_types);
```

to avoid duplicate warnings when there are multiple non-staticlib crate types.